packit_service/worker/handlers/fedmsg_handlers.py

```diff
@@ -124,33 +122,24 @@ def __init__(
             data=data,
         )
         self.branch = data.event_dict.get("branch")
+        self.dg_branch = data.event_dict.get("dg_branch")
 
     def run(self) -> TaskResults:
-        # self.project is dist-git, we need to get upstream
-        dg = DistGit(self.service_config, self.job_config)
-        self.job_config.upstream_project_url = dg.get_project_url_from_distgit_spec()
-        if not self.job_config.upstream_project_url:
-            return TaskResults(
-                success=False,
-                details={
-                    "msg": "URL in specfile is not set. "
-                    "We don't know where the upstream project lives."
-                },
-            )
-
-        n, r = get_namespace_and_repo_name(self.job_config.upstream_project_url)
-        up = self.project.service.get_project(repo=r, namespace=n)
-        self.local_project = LocalProject(
-            git_project=up, working_dir=self.service_config.command_handler_work_dir
+        self.upstream_local_project = LocalProject(
+            git_project=self.project,
+            working_dir=self.service_config.command_handler_work_dir,
+        )
+        self.api = PackitAPI(
+            self.service_config,
+            self.job_config,
+            upstream_local_project=self.local_project,
         )
+        # rev is a commit
+        # we use branch on purpose so we get the latest thing
+        # TODO: check if rev is HEAD on {branch}, warn then?
 
-        self.api = PackitAPI(self.service_config, self.job_config, self.local_project)
         self.api.sync_from_downstream(
-            # rev is a commit
-            # we use branch on purpose so we get the latest thing
-            # TODO: check if rev is HEAD on {branch}, warn then?
-            dist_git_branch=self.branch,
-            upstream_branch=None,  # repo's default branch
+            dist_git_branch=self.dg_branch, upstream_branch=self.branch
```

Comment:
it would be awesome if we could store the result of this action in database.

Reply:
you mean having some table where we would reference the trigger (dg project with the revision), the upstream project, timestamp and so? should I implement it here or later when we implement storing of the projects to sync in db instead of the service config (and discuss the design of this table on arch? )

Reply:
yup, that's exactly what I had in mind, so that we have "an audit trail" of what p-s did and can have a dashboard of those actions

I think that would blow this PR up a lot and we are not doing that now anyway, so let's implement it in the follow-up work.

+1 discussing on the architecture